R/partial.R

```diff
@@ -18,6 +18,18 @@ new_partial <- function(..., class = character()) {
   new_sclr(..., class = c(class, "vctrs_partial"))
 }
 
+## Needed because partial classes inherit from `vctrs_sclr` which
+## can't be renamed. And `vec_ptype2()` etc zap the names.
+#' @export
+`names<-.vctrs_partial` <- function(x, value) {
```

Comment:
This doesn't feel quite right to me — I don't think we should be systematically stripping names, and then providing `names<-` methods that ignore the stripping. Why not just strip names on atomic vectors?

Reply:
The partial vectors are not vectors, which makes them weird in vctrs. This is just a stopgap. I don't think the hacks around partial vectors should make us pause.

As for record vectors, they should support `names<-` at some point, like other vectors. I would really prefer a general solution than ignoring S3 vectors.

Reply:
More generally, maybe there really exists some vectors that do not support names. It doesn't seem ill-founded to require of these vectors to implement set-to-null as a no-op in their `names<-` method.